src/serde_json.rs

```diff
@@ -1156,16 +1156,14 @@ where
     fn ser_json(&self, d: usize, s: &mut SerJsonState) {
         s.out.push('{');
         let len = self.len();
-        let mut index = 0;
-        for (k, v) in self {
+        for (index, (k, v)) in self.iter().enumerate() {
```

Comment:
here as well, I am not quite convinced that there is anything wrong with index += 1

do we really need clippy to be forced on the whole crate?